src/NuGet.Core/NuGet.Protocol.Core.v3/ReportAbuseResourceV3.cs

```diff
@@ -32,50 +33,29 @@ public ReportAbuseResourceV3()
         /// <returns>The first URL from the resource, with the URI template applied.</returns>
         public Uri GetReportAbuseUrl(string id, NuGetVersion version)
         {
-            //return Utility.ApplyPackageIdVersionToUriTemplate(_reportAbuseTemplates.First(), id, version);
-            return new Uri(String.Format(CultureInfo.InvariantCulture, "https://www.nuget.org/packages/{0}/{1}/ReportAbuse", id, version.ToNormalizedString()));
+            try
+            {
+                var uriString = _uriTemplate
+                   .Replace("{id}", id)
+                   .Replace("{version}", version.ToNormalizedString());
+
+                return new Uri(uriString);
+            }
+            catch (UriFormatException)
```

Comment:
Not sure why you will need to catch an exception here, since, there is only one constructor and it ensures that the _uritemplate is assigned a valid value or the default value which is always valid. I will remove the catch clause and add tests if unsure.